src/transformer/utils/get_activity/definition/cmi/choice.php

```diff
@@ -34,40 +34,57 @@
  * @param ?string $description The activity description.
  * @param array $choices The choices available.
  * @param string $lang The language.
+ * @param ?array $rightchoices The correct or sequenced options, optional.
  */
 function choice(
     array $config,
     string $name,
         ?string $description,
     array $choices,
-    string $lang
+    string $lang,
+        ?array $rightchoices = null
 ) {
-    $cmichoices = array_map(
-        function($choice) use ($lang) {
-            return [
-                'id' => utils\slugify($choice),
-                'description' => [
-                    $lang => $choice,
-                ],
-            ];
-        },
-        $choices
-    );
+    $cmichoices = ($config['send_response_choices'] && !is_null($choices))
+        ? array_map(
+            function($choice) use ($lang) {
+                return [
+                    'id' => utils\slugify($choice),
+                    'description' => [
+                        $lang => $choice,
+                    ],
+                ];
+            },
+            $choices
+        )
+        : null;
 
     return [
         ...common($config, $name, $description, $lang),
         'interactionType' => 'choice',
-        'correctResponsesPattern' => [
-            implode(
-                '[,]',
-                array_map(
-                    function($cmichoice) {
-                        return $cmichoice['id'];
-                    },
-                    $cmichoices
-                )
-            ),
-        ],
-        'choices' => $cmichoices
+        ...($config['send_response_choices'] && !empty($cmichoices)
+            ? [
+                'correctResponsesPattern' => [
+                    implode(
+                        '[,]',
+                        // If we are given correct choices, normalize and use
+                        !empty($rightchoices)
+                            ? array_map(
+                                function($rightchoice) {
+                                    return utils\slugify($rightchoice);
+                                },
+                                $rightchoices
+                            )
+                        // Otherwise, just use the choices (no right answer)
```

Comment:
why does this look so weird. Does github not have php comments?

Reply:
it does, but my sick-ass PHP skills have actually broken the syntax highlighting.

Reply:
I think it probably comes down to the unparenthesized test on line 70, those seem kind of optional but who knows